Part 1 of refactor to support button clicks instead of user keyboard input

ETCaton opened the pull request.

Commit:
Part 1 of refactor to support button clicks instead of user keyboard input

ConfusedSky reviewed (changes requested):
Other things from the pause screen branch itself may or may not affect this PR.

* I think the pause menu and the scene select should be separate UI screens, but that can be addressed later. With the pause menu having the ability to go back to the scene select menu
* UIPrefab registry might be better as a map instead of a Vec. 
* Scene buttons should be dynamic based on `scenes::SCENES`
* `SceneSelectState.on_stop` shouldn't silently fail (`is_ok`).
* `SceneSelectState.update` shouldn't be checking to see if the buttons are there every frame, that should be set up in `on_start`. 
* `SceneSelectState.on_start` shouldn't silently fail if pause menu can't be found.
* Maybe also make a button prefab so you don't have to repeat yourself so much. 
* Maybe make the buttons aligned vertically instead of horizontally to fit better with normal game conventions
* Formatting of UI is a little off. Consider changing either the stretch of the root or offsets of each button. 
![image](https://user-images.githubusercontent.com/12839429/85211721-d0a3f780-b300-11ea-9f04-3001333be851.png)

Commit:
Review Comments Changes

Changed initial_state back to impl, added explanatory comments for scene_select's update
